digitaltwins/samples/device/javascript/pnpSampleDevice.js

```diff
@@ -0,0 +1,194 @@
+      propertyUpdateHandler(deviceTwin, componentName, propertyName, null, propertyValue.value, deviceTwin.properties.desired[versionProperty]);
+    });
+  });
+};
+
+
+const sendTelemetryAtInterval = async (deviceClient, componentName, secs) => {
+  let index = 0;
+  const interval = parseFloat(secs)*1000;
+  setInterval( async () => {
+    console.log('Sending telemetry message %d...', index);
+    const data = JSON.stringify({ temp: 1 + (Math.random() * 90), humidity: 1 + (Math.random() * 99) });
+    const pnpMsg = new Message(data);
+    pnpMsg.properties.add(messageSubjectProperty, componentName);
+    pnpMsg.contentType = 'application/json';
+    pnpMsg.contentEncoding = 'utf-8';
+    await deviceClient.sendEvent(pnpMsg);
+    index += 1;
+  }, interval);
+};
+
+async function main() {
+  // fromConnectionString must specify a transport, coming from any transport package.
+  const client = Client.fromConnectionString(deviceConnectionString, Protocol);
+  console.log('Connecting using connection string ' + deviceConnectionString);
+  let resultTwin;
+  try {
+    // Add the modelId here
+    await client.setOptions({ modelId: modelId });
+    await client.open();
+    console.log('Enabling the commands on the client');
+    client.onDeviceMethod(commandNameBlink, commandHandler);
+    client.onDeviceMethod(commandNameTurnOn, commandHandler);
+    client.onDeviceMethod(commandNameTurnOff, commandHandler);
+    // Is there no way to register for an unknown method ?
+
+    // Send Telemetry every 5.5 secs
+    await sendTelemetryAtInterval(client, sensorComponentName, 5.5);
+
+    try {
+      resultTwin = await client.getTwin();
+      const patchDeviceInfo = helperCreateReportedPropertiesPatch({
+        manufacturer: 'Contoso Device Corporation',
+        model: 'Contoso 47-turbo',
+        swVersion: '10.89',
+        osName: 'Contoso_OS',
+        processorArchitecture: 'Contoso_x86',
+        processorManufacturer: 'Contoso Industries',
+        totalStorage: 65000,
+        totalMemory: 640,
+      }, deviceInfoComponentName);
+
+      const patchSDKInfo = helperCreateReportedPropertiesPatch({
+        language: 'node.js',
+        version: packageJson.version,
+        vendor: 'Microsoft'
+      }, sdkInfoComponentName);
+
+      const patchSensorInfo = helperCreateReportedPropertiesPatch({
+        name: 'Node Sample Device',
+        state: true,
+        brightness: '10'
+      }, sensorComponentName);
+
+      // the below things can only happen once the twin is there
+      updateComponentReportedProperties(resultTwin, patchDeviceInfo, deviceInfoComponentName);
+      updateComponentReportedProperties(resultTwin, patchSDKInfo, sdkInfoComponentName);
+      updateComponentReportedProperties(resultTwin, patchSensorInfo, sensorComponentName);
+      helperAttachHandlerForDesiredPropertyPatches(resultTwin, sensorComponentName);
+    } catch (err) {
+      console.error('could not get twin\n' + err.toString());
```

Comment:
>console.error('could not get twin\n' + err.toString()); [](start = 6, length = 55)

Isn't this going to catch the other errors that and happen in the reporting of properties?

Reply:
should be..changed the error message